test/functional/multiwallet.py

```diff
@@ -63,6 +64,10 @@ def run_test(self):
 
         self.start_node(0, self.extra_args[0])
 
+        # ensure listwallets only names BDB files, not symlinks, other files or directories
```

Comment:
I like this new test, but I think it can be improved in a couple of ways:

- test that a non-empty file without the magic bytes is excluded
- there's an implicit assumption that the symlink and folder created earlier in the test are still around. You could explicitly assert that. If the earlier parts of this test get change, eg the file names are changed, then this check will continue to succeed but won't be testing anything.

What do you think about:

```
--- a/test/functional/multiwallet.py
+++ b/test/functional/multiwallet.py
@@ -65,8 +65,20 @@ class MultiWalletTest(BitcoinTestFramework):
         self.start_node(0, self.extra_args[0])
 
         # ensure listwallets only names BDB files, not symlinks, other files or directories
-        open(os.path.join(wallet_dir, 'w01.dat'), 'a').close()
-        assert_equal(set(self.nodes[0].listwallets()['available']), {"w22"})
+        with open(os.path.join(wallet_dir, 'w01.dat'), 'a') as wallet_file:
+            # Create a file containing a long string
+            print("a" * 2000, file=wallet_file)
+
+        available_wallets = self.nodes[0].listwallets()['available']
+
+        assert os.path.islink(os.path.join(wallet_dir, 'w12'))
+        assert 'w12' not in available_wallets
+
+        assert os.path.isdir(os.path.join(wallet_dir, 'w11'))
+        assert 'w11' not in available_wallets
+
+        assert os.path.isfile(os.path.join(wallet_dir, 'w01.dat'))
+        assert 'w01.dat' not in available_wallets
```